src/Functions/FunctionsCoding.h

```diff
@@ -1107,160 +1124,225 @@ class FunctionHex : public IFunction
 
-        if (col_str_in)
-        {
-            auto col_str = ColumnString::create();
-            ColumnString::Chars & out_vec = col_str->getChars();
-            ColumnString::Offsets & out_offsets = col_str->getOffsets();
+             const ColumnString::Chars & in_vec = col_fstr_in->getChars();
 
-            const ColumnString::Chars & in_vec = col_str_in->getChars();
-            const ColumnString::Offsets & in_offsets = col_str_in->getOffsets();
+             size_t size = col_fstr_in->size();
 
-            size_t size = in_offsets.size();
-            out_offsets.resize(size);
-            out_vec.resize(in_vec.size() * 2 - size);
+             out_offsets.resize(size);
+             out_vec.resize(in_vec.size() * word_size + size);
 
-            char * begin = reinterpret_cast<char *>(out_vec.data());
-            char * pos = begin;
-            size_t prev_offset = 0;
+             char * begin = reinterpret_cast<char *>(out_vec.data());
+             char * pos = begin;
 
-            for (size_t i = 0; i < size; ++i)
-            {
-                size_t new_offset = in_offsets[i];
+             size_t n = col_fstr_in->getN();
 
-                executeOneString(&in_vec[prev_offset], &in_vec[new_offset - 1], pos);
+             size_t prev_offset = 0;
 
-                out_offsets[i] = pos - begin;
+             for (size_t i = 0; i < size; ++i)
+             {
+                 size_t new_offset = prev_offset + n;
 
-                prev_offset = new_offset;
-            }
+                 Impl::executeOneString(&in_vec[prev_offset], &in_vec[new_offset], pos);
 
-            if (!out_offsets.empty() && out_offsets.back() != out_vec.size())
-                throw Exception("Column size mismatch (internal logical error)", ErrorCodes::LOGICAL_ERROR);
+                 out_offsets[i] = pos - begin;
+                 prev_offset = new_offset;
+             }
 
-            col_res = std::move(col_str);
+             if (!out_offsets.empty() && out_offsets.back() != out_vec.size())
+                 throw Exception("Column size mismatch (internal logical error)", ErrorCodes::LOGICAL_ERROR);
+
+             col_res = std::move(col_str);
+             return true;
+         }
+         else
+         {
+             return false;
+         }
+     }
+
+    template <typename T>
+    bool tryExecuteFloat(const IColumn * col, ColumnPtr & col_res) const
+    {
+        const ColumnVector<T> * col_vec = checkAndGetColumn<ColumnVector<T>>(col);
+        if (col_vec)
+        {
+            const typename ColumnVector<T>::Container & in_vec = col_vec->getData();
+            Impl::executeFloatAndDecimal(in_vec, col_res, sizeof(T));
             return true;
         }
         else
         {
             return false;
         }
     }
+};
 
-    static bool tryExecuteFixedString(const IColumn * col, ColumnPtr & col_res)
+template <typename Impl>
+class UnConversion : public IFunction
```

Comment:
Too generic name, I've changed it to `DecodeFromBinaryRepr`, but I'm not sure that it also good if you have any other ideas, you can snuggest it.